### PR TITLE
Further memory reductions

### PR DESCRIPTION
**Pull request by ngc92**
With the fused classifier layer, there is actually no need to track logits and dlogits separately, and we can reuse that buffer.
Requires slight changes to the testing script, so we don't test dlogits.
Further, the helper buffers in the attention code aren't actually used during backward, so we really don't need on copy per layer.

### CODE REVIEW

**Comment by karpathy** [comment]
Great finds! I'm seeing memory consumption go 14372MiB -> 10774MiB as a result, saving 3598MiB, **25% of the original**. Huge!
